### DISCUSSION TITLE
How to generate multiple extensions

### DISCUSSION THREAD

**Discussion by nickdrozd** [PyO3/pyo3#3298] [Questions]
I'm porting Python to Rust function-by-function. Currently I have all the ported functions in a module called `rust_stuff`, which is not a great way to organize things! So, is there a way to generate multiple extensions? I guess this would take the form of multiple `.so` files, or something like that.

**Comment by davidhewitt** [comment]
Sorry for the slow reply here. In theory you can do multiple `.so` files, but there are complexities with reusing code, for example see #2340.

Maybe you instead want to organise the functions into submodules within the one `.so`? The `pytests` subdirectory in this repo is a good example of how to do submodules.
